File: app/src/main/java/com/example/cafe/OrderDetailActivity.java
//package com.example.cafe;
//
//import androidx.appcompat.app.AppCompatActivity;
//
//import android.content.Context;
//import android.content.Intent;
//import android.os.Bundle;
//import android.widget.TextView;
//
//
//public class OrderDetailActivity extends AppCompatActivity {
//
//    private static final String EXTRA_DRINK = "drink";
//    private static final String EXTRA_ADDITIVES = "additives";
//    private static final String EXTRA_DRINK_TYPE = "drinkType";
//
//    private TextView textViewName;
//    private TextView textViewDrinks;
//    private TextView textViewDrinkType;
//    private TextView textViewAdditives;
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_order_detail);
//        initView();
//        setupOrder();
//    }
//
//    private void setupOrder() {
//        String drink, additives, drinkType, userName;
//
//        drink = getIntent().getStringExtra(EXTRA_DRINK);
//        additives = getIntent().getStringExtra(EXTRA_ADDITIVES);
//        drinkType = getIntent().getStringExtra(EXTRA_DRINK_TYPE);
//        userName = getIntent().getStringExtra(MakeOrderActivity.EXTRA_USER_NAME);
//
//        textViewName.setText(userName);
//        textViewDrinks.setText(drink);
//        textViewDrinkType.setText(drinkType);
//        textViewAdditives.setText(additives);
//    }
//
//    public static Intent newIntent(Context context,
//                                   String userName,
//                                   String drink,
//                                   String additives,
//                                   String drinkType) {
//
//        Intent intent = new Intent(context, OrderDetailActivity.class);
//        intent.putExtra(MakeOrderActivity.EXTRA_USER_NAME, userName);
//        intent.putExtra(EXTRA_DRINK, drink);
//        intent.putExtra(EXTRA_ADDITIVES, additives);
//        intent.putExtra(EXTRA_DRINK_TYPE, drinkType);
//
//        return intent;
//    }
//
//    private void initView() {
//        textViewDrinks = findViewById(R.id.textViewDrinks);
//        textViewDrinkType = findViewById(R.id.textViewDrinkType);
//        textViewName = findViewById(R.id.textViewName);
//        textViewAdditives = findViewById(R.id.textViewAdditives);
//    }
//}

File: app/src/main/java/com/example/cafe/OrderDetailActivity.kt
package com.example.cafe

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.cafe.databinding.ActivityOrderDetailBinding
import com.example.cafe.Extra.*
class OrderDetailActivity : AppCompatActivity() {

    private lateinit var binding: ActivityOrderDetailBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_order_detail)
        binding = ActivityOrderDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    companion object {
        fun newIntent(
            context: Context,
            userName: String,
            drink: String,
            additives: String,
            drinkType: String
        ): Intent {
            val intent = Intent(context, OrderDetailActivity::class.java)
            with(intent) {
                putExtra(EXTRA_USER_NAME.name, userName)
                putExtra(EXTRA_DRINK.name, drink)
                putExtra(EXTRA_ADDITIVES.name, additives)
                putExtra(EXTRA_DRINK_TYPE.name, drinkType)
                return this
            }
        }
    }
}